frontend/src/pages/editProfile.js

```diff
@@ -1,40 +1,95 @@
-import React from "react";
-import { useSelector, useDispatch } from "react-redux";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 
-import { editProfile } from "../store/profile";
 import InputField from "../components/input";
-import { validate, validateField, getErrorPayload } from "../utils/validation";
+import { validate, validateField } from "../utils/validation";
+import * as profileService from "../services/profileService";
+import { getErrorMessage } from "../utils/validation";
 
 const EditProfile = () => {
-  const state = useSelector((state) => state.editProfile.value);
-  const dispatch = useDispatch();
+  const [state, setState] = useState();
+  const [errors, setErrors] = useState({});
+
   const navigate = useNavigate();
+  let avatarFile = {};
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      const { data } = await profileService.getCurrentProfile();
+      const profile = data.profile;
+      const user = data.user;
+      const payload = {
+        ...state,
+        ...profile,
+        ...user,
+        password: "",
+        password2: "",
+      };
+
+      setState(payload);
```

Comment:
```suggestion
      const { data } = await profileService.getCurrentProfile();

      setState({
        ...data.profile,
        ...data.user,
      });
```
1. no need to create variable if 1 time use
2. i think no need to spread the state since it is initial and no data 
3. password and confirm password can be removed i think 
4. since the profile and user is left and from data, you can also set it directly like this `setState(data);` and call it like this `{state.profile.id}`